src/main/java/com/zendesk/maxwell/bootstrap/MaxwellBootstrapUtility.java

```diff
@@ -22,40 +22,28 @@
 
 	private void run(String[] argv) throws Exception {
 		MaxwellBootstrapUtilityConfig config = new MaxwellBootstrapUtilityConfig(argv);
+
 		if ( config.log_level != null ) {
 			MaxwellLogging.setLevel(config.log_level);
 		}
+
 		ConnectionPool connectionPool = getConnectionPool(config);
 		try ( final Connection connection = connectionPool.getConnection() ) {
-			int rowCount = getRowCount(connection, config);
-			final long rowId = insertBootstrapStartRow(connection, config);
-			Runtime.getRuntime().addShutdownHook(new Thread() {
-				public void run() {
-					try {
-						if ( !isComplete ) {
-							displayLine("");
-							LOGGER.warn("bootstrapping cancelled");
-							removeBootstrapRow(connection, rowId);
-						}
-					} catch ( Exception e ) {
-						System.exit(1);
-					}
-				}
-			});
-			int insertedRowsCount = 0;
-			Long startedTimeMillis = null;
-			while ( !isComplete ) {
-				if ( insertedRowsCount < rowCount ) {
-					if ( startedTimeMillis == null && insertedRowsCount > 0 ) {
-						startedTimeMillis = System.currentTimeMillis();
-					}
-					insertedRowsCount = getInsertedRowsCount(connection, rowId);
-				}
-				isComplete = getIsComplete(connection, rowId);
-				displayProgress(rowCount, insertedRowsCount, startedTimeMillis);
-				Thread.sleep(UPDATE_PERIOD_MILLIS);
+			if ( config.abortBootstrapID != null ) {
+				removeBootstrapRow(connection, config.abortBootstrapID);
+				return;
+			}
+
+			long rowId;
+			if ( config.monitorBootstrapID != null )
+				rowId = config.monitorBootstrapID;
```

Comment:
would be nice to show an error and return if you're trying to monitor a rowId that doesn't exist